modules/core/src/lib/attribute/attribute-transition-manager.ts

```diff
@@ -148,11 +148,9 @@ export default class AttributeTransitionManager {
     // TODO: when switching transition types, make sure to carry over the attribute's
     // previous buffers, currentLength, startIndices, etc, to be used as the starting point
     // for the next transition
-    let needsUpdate = !transition || transition.type !== settings.type;
-    needsUpdate ||=
-      attribute.buffer.byteLength > transition?.attributeInTransition.buffer.byteLength;
```

Comment:
Are you removing this because the buffer length cannot change?

Reply:
I added this line in https://github.com/visgl/deck.gl/pull/8392 thinking that we'd need to re-create the Transition if attribute length changes, but that turned out to be a mistake. When `transition.start()` runs it calls `padBuffer`. Previously that would resize the buffer, now — since buffers have fixed size — it returns a replacement buffer. By keeping the Transition we avoid overwriting older elements in the buffer, which (at least according to the unit tests) is required.